### ISSUE TITLE
Weird behavior

### ISSUE THREAD

**Issue by forcel0** [doxygen/doxygen#10278]
Probably similar to what we were seeing with #10134 that we could not figure out and moved from snippets to small *.cs files and \include.  This worked for the first couple sections but ultimately resurfaced in the .cs files, one listed below.

**Describe the bug**
The same source file (AxisConfigHardwareLimits.cs) when parsed by Doxygen
generateExampleDocs() and then later  generateFileSources() yields html with different output.  generateExampleDocs()  returns _axis_config_hardware_limits_8cs-example.html which is parsed correctly. generateFileSources() stops linking methods during parsing and returns _axis_config_hardware_limits_8cs_source.html which is incorrect.

**Expected behavior**
I would expect the same source file to parse the same

**Screenshots**
If useful, add screenshots to help explain your problem.

**To Reproduce**
It's a large project and if the project is reduced to just the single file the behavior is not present so stack trace is included.

**Version**
Latest 1.9.8 release

**Stack trace**
```
Debug tracing outputs of the AxisConfigHardwareLimits.cs when parsed as an example:
AxisConfigHardware-Example.txt
codifyLines(31,"axis")
generateClassMemberLink(class=RSI::RapidCode::Axis,member=HardwarePosLimitTriggerStateSet)=000001BC8F118B18
type='void' args='(bool state)' class=RSI::RapidCode::Axis
addExampleFile(a5,AxisConfigHardwareLimits.cs,_axis_config_hardware_limits_8cs-example)
stripClassName(scope=RSI::RapidCode::Axis,type=void) classScope=AxisConfigHardwareLimits
stripClass trying 'void' = 0000000000000000
void -> typeName=0000000000000000
---------------------------------------------------------------------------------------
```
Debug tracing outputs of the AxisConfigHardwareLimits.cs when parsed as source:
```
AxisConfigHardware-Source.txt
codifyLines(31,"axis")
codifyLines(31,"HardwarePosLimitTriggerStateSet")
```
It appears that HardwarePosLimitTriggerStateSet is not recognized as a Method to be linked. A different method of the same object is processed correctly 
prior for both at codifyLines(30,"axis").
After the linking breaks at line 31 no further methods are linked though there is a global enum that does.

Readme.txt describes the above and other findings
stylesheets are included to to make the html files easily readable
Doxyfile if it helps in this case with selected options.

[DoxygenTracing.zip](https://github.com/doxygen/doxygen/files/12488955/DoxygenTracing.zip)

**Comment by albert-github** [comment]
The attached file "DoxyTracing.zip"  contains a number of files that might be useful but without a good project it is hard to understand what is going on:
- the Doxyfile does not fit to the, limited, number of source files given
- only the file C# that is included is given
- the html files given don't belong to the same run *2 are from 1.9.8 and 2 are from 1.10.0)
- in the Readme file mentions "GOOD" and "BAD" but what is "GOOD" and what is "BAD"?
- it is unclear how the 2 txt files are generated (which options?)
- it is unclear to what the numbers in Readme.tx (espexcially point 4. refre to

So to be able to see what is wrong we need a complete set and not just some fragments:
- please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by forcel0** [comment]
The Doxyfile was just included for reference of settings.

The html files behave are the same in 1.9.8 and 1.10.0. In order to get tracing enable I had to build from sources so I grabbed the latest sources I can resubmit everything from 1.10.0 if you believe it will help in getting to a solution.

Good = html linking continues as expected
![image](https://github.com/doxygen/doxygen/assets/42810251/7016790a-c69c-4ce5-bcdc-b3c0f6551a52)

Bad = html linking just up and stops even though its the exact same *.cs source file
![image](https://github.com/doxygen/doxygen/assets/42810251/3e547000-f623-485d-916c-b39fd40b5329)

Text files were generated by changing the following in code.l
appears to be where the parser stops linking html
// Toggle for some debugging info
#define DBG_CTX(x) fprintf x
//#define DBG_CTX(x) do { } while(0)

when debugging through code.l 
void CCodeParser::parseCode(OutputCodeList &od,const QCString &className,const QCString &s,
                SrcLangExt lang,bool exBlock, const QCString &exName,const FileDef *fd,
                int startLine,int endLine,bool inlineFragment,
                const MemberDef *memberDef,bool showLineNumbers,const Definition *searchCtx,
                bool collectXRefs)

The line numbers are of the source file being parsed. In this case the included AxisConfigHardwareLimits.cs file. Line 31 of AxisConfigHardwareLimits.cs is where Doxygen stops hyperlinking methods of the class for source but the same AxisConfigHardwareLimits.cs file is parsed and linked correctly when parsing for examples.

**Comment by albert-github** [comment]
It is a bit cleared now, problem is that in the GOOD part there is linking and the BAD there is not.
Most likely due to other included parts, but to be able to check what is going wrong we need the example so 
- please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

Rerunning everything with the current master is not necessary, the small but full example is necessary so we can try to reproduce the problem.

**Comment by forcel0** [comment]
I'll see if I can put a sample together.

**Comment by albert-github** [comment]
A commit ([13c270e](https://github.com/doxygen/doxygen/commit/13c270eedb4128251a1ca2efbe4d4d319c6d90ec)) has been added to the master version of doxygen introducing a new option `local` for the `\include` command.

@forcel0 does this, help to, solve your problem?

@doxygen I see that the `local` is only documented for the `\include` command, but seen the code, `if .. else if`, structure does this also influence the `\snippet` command?

**Comment by doxygen** [comment]
@albert-github It also applies to the `\snippet` command indeed, but its documentation was also updated (see line 2840 of commands.doc).

**Comment by albert-github** [comment]
@doxygen Sorry I missed that change with `\snippet`.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by forcel0** [comment]
I did check with with 1.10.0 version in an earlier included example, but I will sanity check with the official release and will send an example and description if we still see an issue.ThankOn Dec 25, 2023, at 6:17 AM, Dimitri van Heesch ***@***.***> wrote:﻿
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
